pkg/daemons/config/types.go

```diff
@@ -136,6 +136,7 @@ type CriticalControlArgs struct {
 	DisableHelmController bool
 	DisableNPC            bool
 	DisableServiceLB      bool
+	SecretsEncrypt        bool
```

Comment:
I am pretty sure that changing this will affect the serialized representation of the cluster configuration, leading to confusion when trying to join clusters of differing versions.

```console
brandond@dev01:~$ curl -ks https://node:token@172.17.0.2:6443/v1-k3s/config | jq . | grep Encrypt
  "EncryptSecrets": false,
  "EncryptForce": false,
  "EncryptSkip": false,
```


You might be able to get around this by overriding the serialized field name:
```suggestion
	SecretsEncrypt        bool `json:"EncryptSecrets"`
```

Reply:
Yeah that works... I want to change the name because of the warning gets weird with 
```
level=warning msg="critical configuration EncryptSecrets is mismatched"
```

Reply:
Validated that this works
```
curl -ks https://node:12345@127.0.0.1:6443/v1-k3s/config | jq . | grep Encrypt
  "EncryptSecrets": true,
  "EncryptForce": false,
  "EncryptSkip": false,
```